src/geocoder.js

```diff
@@ -35,12 +40,23 @@ module.exports = function(url, options) {
         }
 
         feedback.record({geocoding: query});
-        return L.Util.template(geocoder.getURL(), {query: query});
+
+        return L.Util.template(geocoder.getURL(), {
+            proximity: args.proximity ? encodeURIComponent([args.proximity.lng, args.proximity.lat]) : '',
+            query: query 
+        });
     };
 
     geocoder.query = function(_, callback) {
         util.strict(callback, 'function');
-        request(geocoder.queryURL(_), function(err, json) {
+
+        var args = {};
+        if (typeof _ === 'object' && !Array.isArray(_)) {
```

Comment:
[`Array.isArray` is not supported by IE8](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#Browser_compatibility), which Mapbox.js supports. I suggest reversing the condition and testing for a string instead.